.github/workflows/pullrequest.yml

```diff
@@ -108,28 +125,31 @@ jobs:
         if: steps.node-modules.outputs.cache-hit != 'true'
         run: ./scripts/ci/10_prepare-host-deps.sh
 
-      - run: |
+      - name: Preparing HEAD and BASE tags
+        continue-on-error: true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          HTML_URL: ${{ github.event.pull_request.html_url }}
+          ISSUE_REPORTING_SLACK_WEBHOOK_URL: ${{ secrets.SLACK_BUILD_ISSUES_REPORTING_WEBHOOK_URL }}
+        run: |
+          set -x
+          set -euo pipefail
           echo "HEAD=$GITHUB_SHA" >> $GITHUB_ENV
           export HEAD_REF="$GITHUB_HEAD_REF"
           export BASE_REF="$GITHUB_BASE_REF"
-          export PR_REF=$GITHUB_SHA
+          export PR_REF="$GITHUB_SHA"
           export SHELL=/usr/bin/bash
           export WORKFLOW_ID=pullrequest
-          source ./scripts/ci/00_prepare-base-tags.sh $(git merge-base HEAD $GITHUB_BASE_REF)
-          git checkout $GITHUB_SHA
+          source ./scripts/ci/00_prepare-base-tags.sh $(git merge-base HEAD "$GITHUB_BASE_REF")
+          git checkout "$GITHUB_SHA" # GITHUB_SHA
           echo "BASE=$BASE" >> $GITHUB_ENV
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          HTML_URL: ${{ github.event.pull_request.html_url }}
-          ISSUE_REPORTING_SLACK_WEBHOOK_URL: ${{ secrets.SLACK_BUILD_ISSUES_REPORTING_WEBHOOK_URL }}
-        name: Preparing HEAD and BASE tags
 
       - name: Cache for cypress
         id: cypress
         continue-on-error: true
         uses: ./.github/actions/cache
         with:
-          path: /github/home/.cache/Cypress
+          path: ~/.cache/Cypress
```

Comment:
Are you sure ~ works in github actions?
Has this been tested?

Reply:
It passes locally when testing with `act`, and passes in _this_ PR